File: nodejs/src/factory/simple_factory.js
const Food = require('./food');

function getFood(foodType, foodName) {
  if (!Food[foodType]) {
    throw new Error(`Invaild food type: ${foodType}`);
  }

  return new Food[foodType](foodName);
}

const food = getFood('bbq', 'BBQ');
console.log(food.constructor);
console.log(food.getName());

File: python/src/factory/simpleFactory.py
class BBQ:
    def __init__(self, name):
        self.name = name

    def cook(self):
        return {
            'name': self.name,
            'type': 'BBQ'
        }

class Stew:
    def __init__(self, name):
        self.name = name

    def cook(self):
        return {
            'name': self.name,
            'type': 'Stew'
        }

class Drink:
    def __init__(self, name):
        self.name = name

    def cook(self):
        return {
            'name': self.name,
            'type': 'Drink'
        }

def factory(foodType):
    foodTypes = {
        'BBQ': BBQ,
        'Stew': Stew,
        'Drink': Drink
    }
    return foodTypes[foodType]

Food = factory('BBQ')
food = Food('bot')
print(food.cook())
